Refactor `active` (#869)
calliope-project/calliope: docs/user_defined_math/components.md

These are not required, but are useful metadata for later reference.
1. It can have a top-level `foreach` list and `where` string.
Without a `foreach`, it becomes an un-indexed variable.

> Without a `where` string, all valid members (according to the `active`) based on `foreach` will be included in this decision variable.

Reviewer: Small tuning so that this reads a bit more natural.

```suggestion
Without a `where` string, all `active` members in `foreach` will be included in this decision variable.
```

These are not required, but are useful metadata for later reference.
1. It can have a top-level `foreach` list and `where` string.
Without a `foreach`, it becomes an un-indexed expression.

> Without a `where` string, all valid members (according to the `active`) based on `foreach` will be included in this expression.

Reviewer: Another small tuning for readability.

```suggestion
Without a `where` string, all `active` members in `foreach` will be included in this decision variable.
```

These are not required, but are useful metadata for later reference.
1. It can have a top-level `foreach` list and `where` string.
Without a `foreach`, it becomes an un-indexed constraint.

> Without a `where` string, all valid members (according to the `active`) based on `foreach` will be included in this constraint.

Reviewer: ```suggestion
Without a `where` string, all `active` members in `foreach` will be included in this decision variable.
```

This is not required, but is useful metadata for later reference.
1. It can have a top-level `foreach` list and `where` string.
Without a `foreach`, it becomes an un-indexed constraint.

> Without a `where` string, all valid members (according to the `active`) based on `foreach` will be included in this constraint.

Reviewer: ```suggestion
Without a `where` string, all `active` members in `foreach` will be included in this decision variable.
```

   These are not required, but are useful metadata for later reference.
1. It can have a top-level `foreach` list and `where` string.
   Without a `foreach`, it becomes an un-indexed expression.

>    Without a `where` string, all valid members (according to the `active`) based on `foreach` will be included in this expression.

Reviewer: ```suggestion
   Without a `where` string, all `active` members in `foreach` will be included in this decision variable.
```
